clang-tools-extra/test/clang-tidy/checkers/bugprone/forwarding-reference-overload.cpp

```diff
@@ -251,3 +251,10 @@ class Test10 {
   Test10(T &&Item, E e)
       : e(e){}
 };
+
+// A deleted ctor cannot hide anything
+class Test11 {
+public:
+  template <typename T>
+  Test11(T&&) = delete;
```

Comment:
Although this is enough to trigger (or not after the fix) the issue, perhaps I should include a copy+move ctor below for clarity?